src/data/finsupp/basic.lean

```diff
@@ -280,6 +280,29 @@ lemma single_left_inj (h : b ≠ 0) :
   and_false, or_false, eq_self_iff_true, and_true] using H,
  λ H, by rw [H]⟩
 
+lemma single_support_ne_bot (i : α) (h : b ≠ 0) :
+  (single i b).support ≠ ⊥ :=
+begin
+  have : i ∈ (single i b).support := by simpa using h,
+  intro H,
+  simpa [H]
+end
+
+lemma single_support_disjoint {b'} (hb : b ≠ 0) (hb' : b' ≠ 0) {i j : α} :
+  disjoint (single i b).support (single j b').support ↔ i ≠ j :=
+begin
+  split,
+  { rintro H rfl,
+    have : i ∈ (single i b).support ⊓ (single i b').support,
+      { simp [hb, hb'] },
+    simpa using H this },
+  { intros hne x,
+    suffices : i = x → j = x → x ∈ ⊥,
+      { simpa [single_apply, hb, hb'] },
+    rintro rfl rfl,
+    simpa using hne }
+end
```

Comment:
Do we have a proof that `finsupp.support f = function.support (coe_fn f)`?

If so, how hard is it to move this proof to be about `function.support`, and then do the hopefully small translation here?